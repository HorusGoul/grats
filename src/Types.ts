/** @gqlScalar */
export type Float = number;
/** @gqlScalar */
export type Int = number;
