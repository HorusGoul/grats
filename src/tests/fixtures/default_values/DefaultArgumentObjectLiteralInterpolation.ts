import { Int } from "../../../Types";

const first = 10;
const offset = 100;
/** @gqlType */
export default class Query {
  /** @gqlField */
  someField1({
    input = { first, offset },
  }: {
    input?: ConnectionInput;
  }): string {
    return "hello";
  }
}

/** @gqlInput */
type ConnectionInput = {
  first: Int;
  offset: Int;
};
