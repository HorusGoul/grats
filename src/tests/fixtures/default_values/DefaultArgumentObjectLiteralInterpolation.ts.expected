-----------------
INPUT
----------------- 
import { Int } from "../../../Types";

const first = 10;
const offset = 100;
/** @gqlType */
export default class Query {
  /** @gqlField */
  someField1({
    input = { first, offset },
  }: {
    input?: ConnectionInput;
  }): string {
    return "hello";
  }
}

/** @gqlInput */
type ConnectionInput = {
  first: Int;
  offset: Int;
};

-----------------
OUTPUT
-----------------
src/tests/fixtures/default_values/DefaultArgumentObjectLiteralInterpolation.ts:9:15 - error: Expected object literal property to be a property assignment.

If you think Grats should be able to infer this type, please report an issue at https://github.com/captbaritone/grats/issues.

9     input = { first, offset },
                ~~~~~
src/tests/fixtures/default_values/DefaultArgumentObjectLiteralInterpolation.ts:9:22 - error: Expected object literal property to be a property assignment.

If you think Grats should be able to infer this type, please report an issue at https://github.com/captbaritone/grats/issues.

9     input = { first, offset },
                       ~~~~~~
